### DISCUSSION TITLE
Lubuntu + xrdp: FeatherPad won't launch

### DISCUSSION THREAD

**Discussion by mtvessel** [tsujan/FeatherPad#620] [Q&A]
FeatherPad works fine when connecting locally to my Lubuntu 20.04 instance,   When connected from Windows 10, using RDP (xrdp on Lubuntu), all other lxqt apps work, but FeatherPad doesn't launch.

Sorry for the vague description, but please let me know what other details I can provide.  And thank you for this very useful text editor!

**Comment by tsujan** [comment]
I converted your report to a discussion because the problem you've encountered can hardly be in FeatherPad.

I'm not familiar with RDP or `xrdp`, but there may be a message somewhere. Run FeatherPad from a terminal to see if there is any message.

Also, as far as I know, the version of FeatherPad in Lubuntu should be very old, although that may not be relevant in this case.

**Reply by mtvessel** [reply]
Yes, it doesn't seem like it would be a FeatherPad issue, but it is odd that this seems to be the only bundled app that won't launch.  I hadn't realized the apt-get sources weren't synced with your releases.  I'll try to update my install.  Thanks!

**Reply by tsujan** [reply]
I'm afraid you didn't answer the main question about a message in the terminal. There can be many reasons why an app doesn't start and, without a minimum amount of info, no one could find the cause.

**Reply by mtvessel** [reply]
Apologies, I did some googling, and ended up tweaking my /etc/xrdp/startwm.sh.  After a reboot, FeatherPad works fine over rdp now.  I'm sorry for the red herring, and again, thanks for a great utility.

Oh, for completeness, there was no error message, originally, when launched from the terminal.  The shell accepted the command, returned, but no FeatherPad appeared.

**Reply by tsujan** [reply]
No need to an apology. You had a problem that somehow involved FeatherPad. The Discussions section is for Q&A and more.

Also, you might want to be more specific about what you did to `/etc/xrdp/startwm.sh`. It might be helpful to someone with the same problem.

**Reply by mtvessel** [reply]
Well, I made many changes along the way, and I'm honestly not sure what did the trick here, including switching from lxqt to xfce and back again (FeatherPad did not work under xfce either).  The only change that seems remotely related, is that in `/etc/xrdp/startwm.sh`, I added the line

```. $HOME/.profile```

right before the line:

```if test -r /etc/profile; then```

based on advice from this [youtube video - How to Fix Black Screen When Remote Desktop to Ubuntu Desktop 20 from Windows 10 Using XRDP](https://www.youtube.com/watch?v=0PlG6vCOmjc).

Cheers!

**Reply by tsujan** [reply]
Thanks!

As I said above, I'm not familiar with `xrdp`, but I still don't understand how it's possible that no message was shown in the terminal. Apparently, an RDP session has its own peculiarities.

**Reply by tsujan** [reply]
Oh, one simple explanation comes to my mind:

Maybe, a FeatherPad window was already open in another desktop, so that it wasn't visible. If RDP had a problem in recognizing X11 virtual desktops, each `featherpad` command would just open a new tab in that window and return, and since the window wasn't visible, the user would think that nothing happened.

**Reply by mtvessel** [reply]
I don't know much about xrdp myself (other than as a user), but that's quite plausible.  Previously, I had a session open at the console,  and it may have been still logged in when I was first trying rdp.  After the reboot, I did not log in at the console, and it worked without issue.

**Reply by tsujan** [reply]
You could also check `Preferences → Window → Always open in separate windows` when using RDP. It causes new files to be opened in new windows, instead of new tabs of the same window.
